lib/Sema/TypeCheckDecl.cpp

```diff
@@ -810,6 +810,13 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
         }
       }
 
+      // Skip conflicts with inherited initializers, which only happen when the
+      // current declaration is within an extension. The override checker should
+      // have already taken care of emitting a more productive diagnostic.
+      if (isa<ConstructorDecl>(other) && other->isImplicit() &&
+          other->getOverriddenDecl())
+        continue;
```

Comment:
I'm concerned by this not requiring any changes to existing tests, and I don't think your new test covers it. What if the inherited initializer *is* exposed to Objective-C?

Reply:
Also, I'm not sure exiting early is a good idea, because we still need to mark one of them invalid.

Reply:
Sema, Constraints, decl and Generics pass just fine, but it might affect something elsewhere. 

> and I don't think your new test covers it. What if the inherited initializer is exposed to Objective-C?

You're right, I'll add some more tests with `@objc` and `dynamic`. Right now though it boils down to `'init(arg:)' has already been overridden` with and without the change if I plug all the errors.